clean_code/player.py

```diff
@@ -1,142 +1,153 @@
-        pausing = False
-        label1['text'] = song_data()
-    except:
-        pass
-#################################################################################
-def check_music():
-    global list_index, list1, SONG_END, pausing, label1, text1
-    try:
-        for event in pygame.event.get():
-            if event.type == SONG_END:
-                button5()
-    except:
-        pass
-#################################################################################
-def button3():
-    global list_index, list1, SONG_END, pausing, label1, text1
-    try:
-        if pausing:
+
+        self.list_of_songs = []
+        self.paused = False
+        self.current_song_index = 0
+
+        self.song_info_label = None
+        self.song_list_text = None
+
+        self.create_widgets()
+
+    def create_widgets(self):
+        self.create_buttons()
+        self.create_song_info_label()
+        self.create_song_list_text()
+
+    def create_buttons(self):
+        BUTTONS = [
+            ("ADD TO LIST", self.add_songs_to_list, self.BUTTONS_ROW[0]),
+            ("PLAY SONG", self.play_song, self.BUTTONS_ROW[1]),
+            ("PAUSE/UNPAUSE", self.toggle_pause, self.BUTTONS_ROW[2]),
+            ("PREVIOUS SONG", self.play_previous_song, self.BUTTONS_ROW[3]),
+            ("NEXT SONG", self.play_next_song, self.BUTTONS_ROW[4]),
+        ]
+
+        for text, command, row in BUTTONS:
+            self.create_button(text, command, row)
+
+    def create_button(self, text, command, row):
+        Button(self, text=text, command=command, bg=self.BUTTON_BG_COLOR, width=self.BUTTON_WIDTH).grid(row=row, column=self.COLUMN)
+
+    def create_song_info_label(self):
+        self.song_info_label = Label(self, fg='black', font=self.FONT_STYLE, bg=self.LABEL_BG_COLOR)
+        self.song_info_label.grid(row=self.SONG_INFO_ROW, column=self.COLUMN)
+
+    def create_song_list_text(self):
+        self.song_list_text = Text(self, wrap=WORD, width=self.TEXT_WIDTH)
+        self.song_list_text.grid(row=self.SONG_LIST_ROW, column=self.COLUMN)
+
+    def add_songs_to_list(self):
+        selected_files = askopenfilenames()
+        self.list_of_songs.extend(selected_files)
+        self.update_song_list_text()
+
+    def update_song_list_text(self):
+        self.song_list_text.delete(0.0, 'end')
+
+        for index, song_file in enumerate(self.list_of_songs, start=1):
+            self.song_list_text.insert('end', f"{index}: {self.get_song_data(song_file)}\n")
+
+    @staticmethod
+    def get_song_data(song_file):
+        try:
+            song = EasyID3(song_file)
+            title = song.get('title', [MusicPlayer.DEFAULT_SONG_TITLE])[0]
+            artist = song.get('artist', [MusicPlayer.DEFAULT_SONG_ARTIST])[0]
+            return f"{title} - {artist}"
+        except Exception as error:
+            print(f"Could not get song data {error}")
+            return f"{MusicPlayer.DEFAULT_SONG_TITLE} - {MusicPlayer.DEFAULT_SONG_ARTIST}"
```

Comment:
instead of having this return logic twice, if you want to catch the EastID3 error, then you can just wrap it in try except clause, and in the except case set the default `song` dict to your default values. This way you will avoid this code repetition